src/Altinn.App.Core/Internal/Expressions/LayoutEvaluatorStateInitializer.cs

```diff
@@ -9,34 +12,101 @@ namespace Altinn.App.Core.Internal.Expressions;
 /// <summary>
 /// Utility class for collecting all the services from DI that are needed to initialize <see cref="LayoutEvaluatorState" />
 /// </summary>
-public class LayoutEvaluatorStateInitializer
+public class LayoutEvaluatorStateInitializer : ILayoutEvaluatorStateInitializer
 {
     // Dependency injection properties (set in ctor)
     private readonly IAppResources _appResources;
+    private readonly IAppMetadata _appMetadata;
     private readonly FrontEndSettings _frontEndSettings;
 
     /// <summary>
     /// Constructor with services from dependency injection
     /// </summary>
-    public LayoutEvaluatorStateInitializer(IAppResources appResources, IOptions<FrontEndSettings> frontEndSettings)
+    public LayoutEvaluatorStateInitializer(
+        IAppResources appResources,
+        IAppMetadata appMetadata,
+        IOptions<FrontEndSettings> frontEndSettings
+    )
     {
         _appResources = appResources;
+        _appMetadata = appMetadata;
         _frontEndSettings = frontEndSettings.Value;
     }
 
+    /// <summary>
+    /// Helper class to keep compatibility with old interface
+    /// delete when <see cref="LayoutEvaluatorStateInitializer.Init(Altinn.Platform.Storage.Interface.Models.Instance,object,string?,string?)"/>
+    /// is removed
+    /// </summary>
+    private sealed class SingleDataElementAccessor : IInstanceDataAccessor
+    {
+        private readonly DataElement _dataElement;
+        private readonly object _data;
+
+        public SingleDataElementAccessor(Instance instance, DataElement dataElement, object data)
+        {
+            Instance = instance;
+            _dataElement = dataElement;
+            _data = data;
+        }
+
+        public Instance Instance { get; }
+
+        public Task<object> GetData(DataElementId dataElementId)
+        {
+            if (dataElementId != _dataElement)
+            {
+                return Task.FromException<object>(
+                    new InvalidOperationException(
+                        "Use the new ILayoutEvaluatorStateInitializer interface to support multiple data models and subforms"
+                    )
+                );
+            }
+            return Task.FromResult(_data);
+        }
+
+        public Task<object?> GetSingleDataByType(string dataType)
+        {
+            if (_dataElement.DataType != dataType)
+            {
+                return Task.FromException<object?>(
+                    new InvalidOperationException("Data type does not match the data element")
+                );
+            }
+            return Task.FromResult<object?>(_data);
+        }
+    }
+
     /// <summary>
     /// Initialize LayoutEvaluatorState with given Instance, data object and layoutSetId
     /// </summary>
-    public virtual Task<LayoutEvaluatorState> Init(
+    [Obsolete("Use the overload with ILayoutEvaluatorStateInitializer instead")]
+    public async Task<LayoutEvaluatorState> Init(
```

Comment:
So if I understand correctly this and the code above is kept for compatibility, should we still have some tests for this to make sure it still works? From what I can tell it's only SSB that are using this to acccess `RemoveHiddenData`, so maybe just a test for that?

Reply:
Added in aee32775a664c5117c2fd9f9bb5639f8fcea1de9